orchestration/docker.go

```diff
@@ -3,104 +3,141 @@ package orchestration
 import (
 	"encoding/json"
 	"fmt"
+	"log"
 	"os"
 	"os/exec"
-	"regexp"
 	"strings"
 
+	"github.com/AudiusProject/audius-d/conf"
 	"github.com/joho/godotenv"
 )
 
-func runNodeDocker(nodeType string, network string, imageTag string, autoUpgrade bool) {
-	fmt.Printf("standing up %s on network %s\n", nodeType, network)
-	volumeFlag := ""
-	//volumeFlag = fmt.Sprintf("-v %s:/root/audius-docker-compose/%s/override.env", confFilePath, nodeType)
-
-	if !regexp.MustCompile(`^[a-zA-Z0-9_\-]+$`).MatchString(imageTag) {
-		exitWithError("Invalid image tag:", imageTag)
-	}
-
-	// volume create is idempotent
-	if err := runCommand("/bin/sh", "-c", "docker volume create audius-d"); err != nil {
-		exitWithError("Error executing command:", err)
-	}
-
-	var cmd string
-	baseCmd := fmt.Sprintf(`docker run --privileged -d -v audius-d:/var/lib/docker %s -p %d:80 -p %d:443`, volumeFlag, 80, 443)
-
-	switch nodeType {
-	case "creator-node":
-		cmd = fmt.Sprintf(baseCmd + ` \
-        --name creator-node \
-        -v /var/k8s/mediorum:/var/k8s/mediorum \
-        -v /var/k8s/creator-node-backend:/var/k8s/creator-node-backend \
-        -v /var/k8s/creator-node-db:/var/k8s/creator-node-db \
-        audius/audius-docker-compose:` + imageTag)
-	case "discovery-provider":
-		baseCmd = baseCmd + " -p 5000:5000"
-		cmd = fmt.Sprintf(baseCmd + ` \
-        --name discovery-provider \
-        -v /var/k8s/discovery-provider-db:/var/k8s/discovery-provider-db \
-        -v /var/k8s/discovery-provider-chain:/var/k8s/discovery-provider-chain \
-        audius/audius-docker-compose:` + imageTag)
-	case "identity-service":
-		baseCmd = baseCmd + " -p 7000:7000"
-		cmd = fmt.Sprintf(baseCmd + ` \
-        --name identity-service \
-        audius/audius-docker-compose:` + imageTag)
-	default:
-		exitWithError(fmt.Sprintf("provided node type is not supported: %s", nodeType))
+type OverrideEnv = map[string]string
+
+// deploys a server node regardless of type
+func RunNode(config conf.ContextConfig, serverConfig conf.BaseServerConfig, override OverrideEnv, containerName string, nodeType string, internalVolumes []string) error {
```

Comment:
I don't have super strong feelings about this, but maybe instead of passing in the ContextConfig, just pass in the NetworkConfig - currently it looks like you'll only using config.Network anyway, and it's probably also better if this function can work with as little information as possible.